documentation/prototypes.md

```diff
@@ -0,0 +1,74 @@
+# Prototype process
```

Comment:
Nothing new, just a move of file from https://github.com/meilisearch/engine-team to here